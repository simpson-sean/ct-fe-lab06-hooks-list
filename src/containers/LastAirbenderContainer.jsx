import React, {useState, useEffect} from 'react';
import { findCharacters } from '../services/LastAirbenderAPI';
import CharacterList from '../components/characters/CharacterList';

const LastAirbenderContainer = () => {
    
    const [loading, setLoading] = useState(true);
    const [characters, setCharacters] = useState([])

    useEffect( () => {
        findCharacters().then((characters) => {
            setCharacters(characters);
            setLoading(false);
        })
    }, [] )

    if(loading)return<img src="https://powerusers.microsoft.com/t5/image/serverpage/image-id/118082i204C32E01666789C?v=v2"
    alt="loading spinner" />
            
    return(
        <CharacterList characters={characters} />
        )
};

export default LastAirbenderContainer;