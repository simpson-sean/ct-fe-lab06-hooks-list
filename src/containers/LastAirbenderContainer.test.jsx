import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import listData from '../fixtures/list.json';
import LastAirbenderContainer from './LastAirbenderContainer';

const server = setupServer(
  rest.get('https://last-airbender-api.herokuapp.com/api/v1/characters', (req, res, ctx) => {
    return res(ctx.json(listData));
  })
);

describe('LastAirbender Container', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('displays a list of characters on the page', async () => {
    const { container } = render(
      <MemoryRouter>
        <LastAirbenderContainer />
      </MemoryRouter>
    );

    screen.getByAltText("loading spinner");

    const ul = await screen.findByRole('list', { name: 'characters' });
    expect(ul).not.toBeEmptyDOMElement();
    expect(container).toMatchSnapshot();
  });
});