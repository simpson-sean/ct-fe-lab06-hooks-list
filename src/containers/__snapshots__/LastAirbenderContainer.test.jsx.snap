// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LastAirbender Container displays a list of characters on the page 1`] = `
<div>
  <ul
    aria-label="characters"
  >
    <div>
      <li>
        <a
          href="/5cf5679a915ecad153ab68c8"
        >
          <img
            src="https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441"
          />
          <h1 />
          <h3 />
          <h3 />
        </a>
      </li>
    </div>
    <div>
      <li>
        <a
          href="/5cf5679a915ecad153ab68c9"
        >
          <img
            src="https://vignette.wikia.nocookie.net/avatar/images/a/ae/Aang_at_Jasmine_Dragon.png/revision/latest?cb=20130612174003"
          />
          <h1 />
          <h3 />
          <h3 />
        </a>
      </li>
    </div>
    <div>
      <li>
        <a
          href="/5cf5679a915ecad153ab68ca"
        >
          <img
            src="https://vignette.wikia.nocookie.net/avatar/images/2/21/Into_The_Inferno_Aang.png/revision/latest?cb=20131009060746"
          />
          <h1 />
          <h3 />
          <h3 />
        </a>
      </li>
    </div>
    <div>
      <li>
        <a
          href="/5cf5679a915ecad153ab68cb"
        >
          <img
            src="https://vignette.wikia.nocookie.net/avatar/images/7/79/Pilot_-_Aang.png/revision/latest?cb=20120311133235"
          />
          <h1 />
          <h3 />
          <h3 />
        </a>
      </li>
    </div>
    <div>
      <li>
        <a
          href="/5cf5679a915ecad153ab68cc"
        >
          <img
            src="https://vignette.wikia.nocookie.net/avatar/images/2/24/Afiko.png/revision/latest?cb=20121121024128"
          />
          <h1 />
          <h3 />
          <h3 />
        </a>
      </li>
    </div>
    <div>
      <li>
        <a
          href="/5cf5679a915ecad153ab68cd"
        >
          <img
            src="https://vignette.wikia.nocookie.net/avatar/images/c/cd/Air_Acolyte_woman.png/revision/latest?cb=20140421100225"
          />
          <h1 />
          <h3 />
          <h3 />
        </a>
      </li>
    </div>
    <div>
      <li>
        <a
          href="/5cf5679a915ecad153ab68ce"
        >
          <img
            src="https://vignette.wikia.nocookie.net/avatar/images/7/79/Air_Nomad_boy.png/revision/latest?cb=20130714154218"
          />
          <h1 />
          <h3 />
          <h3 />
        </a>
      </li>
    </div>
    <div>
      <li>
        <a
          href="/5cf5679a915ecad153ab68cf"
        >
          <img
            src="https://vignette.wikia.nocookie.net/avatar/images/0/07/Aiwei_reveals_Korra%27s_lie.png/revision/latest?cb=20140808130551"
          />
          <h1 />
          <h3 />
          <h3 />
        </a>
      </li>
    </div>
    <div>
      <li>
        <a
          href="/5cf5679a915ecad153ab68d0"
        >
          <img
            src="https://vignette.wikia.nocookie.net/avatar/images/0/00/Amon_and_his_Equalists.png/revision/latest?cb=20160219113422"
          />
          <h1 />
          <h3 />
          <h3 />
        </a>
      </li>
    </div>
    <div>
      <li>
        <a
          href="/5cf5679a915ecad153ab68d1"
        >
          <img
            src="https://vignette.wikia.nocookie.net/avatar/images/c/c3/Analay.png/revision/latest?cb=20140430182201"
          />
          <h1 />
          <h3 />
          <h3 />
        </a>
      </li>
    </div>
    <div>
      <li>
        <a
          href="/5cf5679a915ecad153ab68d2"
        >
          <img
            src="https://vignette.wikia.nocookie.net/avatar/images/5/58/Animal_control_officer.png/revision/latest?cb=20140504160655"
          />
          <h1 />
          <h3 />
          <h3 />
        </a>
      </li>
    </div>
    <div>
      <li>
        <a
          href="/5cf5679a915ecad153ab68d3"
        >
          <img
            src="https://vignette.wikia.nocookie.net/avatar/images/6/65/Appa_flying.png/revision/latest?cb=20140517110636"
          />
          <h1 />
          <h3 />
          <h3 />
        </a>
      </li>
    </div>
    <div>
      <li>
        <a
          href="/5cf5679a915ecad153ab68d4"
        >
          <img
            src="https://vignette.wikia.nocookie.net/avatar/images/b/be/Appa%27s_mother.png/revision/latest?cb=20130705194428"
          />
          <h1 />
          <h3 />
          <h3 />
        </a>
      </li>
    </div>
    <div>
      <li>
        <a
          href="/5cf5679a915ecad153ab68d5"
        >
          <img
            src="https://vignette.wikia.nocookie.net/avatar/images/6/6f/Arik.png/revision/latest?cb=20140808184002"
          />
          <h1 />
          <h3 />
          <h3 />
        </a>
      </li>
    </div>
    <div>
      <li>
        <a
          href="/5cf5679a915ecad153ab68d6"
        >
          <img
            src="https://vignette.wikia.nocookie.net/avatar/images/0/0c/Chief.png/revision/latest?cb=20140122221730"
          />
          <h1 />
          <h3 />
          <h3 />
        </a>
      </li>
    </div>
    <div>
      <li>
        <a
          href="/5cf5679a915ecad153ab68d7"
        >
          <img
            src="https://vignette.wikia.nocookie.net/avatar/images/b/b6/Asami_Sato.png/revision/latest?cb=20150407110150"
          />
          <h1 />
          <h3 />
          <h3 />
        </a>
      </li>
    </div>
    <div>
      <li>
        <a
          href="/5cf5679a915ecad153ab68d8"
        >
          <img
            src="https://vignette.wikia.nocookie.net/avatar/images/9/9a/Ashuna.png/revision/latest?cb=20161026183558"
          />
          <h1 />
          <h3 />
          <h3 />
        </a>
      </li>
    </div>
    <div>
      <li>
        <a
          href="/5cf5679a915ecad153ab68d9"
        >
          <img
            src="https://vignette.wikia.nocookie.net/avatar/images/7/76/Aye-aye_spirit.png/revision/latest?cb=20131019224533"
          />
          <h1 />
          <h3 />
          <h3 />
        </a>
      </li>
    </div>
    <div>
      <li>
        <a
          href="/5cf5679a915ecad153ab68da"
        >
          <img
            src="https://vignette.wikia.nocookie.net/avatar/images/1/12/Azula.png/revision/latest?cb=20140905084941"
          />
          <h1 />
          <h3 />
          <h3 />
        </a>
      </li>
    </div>
    <div>
      <li>
        <a
          href="/5cf5679a915ecad153ab68db"
        >
          <img
            src="https://vignette.wikia.nocookie.net/avatar/images/a/ad/Azula_%28games%29.png/revision/latest?cb=20131228015042"
          />
          <h1 />
          <h3 />
          <h3 />
        </a>
      </li>
    </div>
  </ul>
</div>
`;
